crates/metadata/lib.rs

```diff
@@ -482,6 +482,36 @@ mod test_parsing {
         "#;
         let metadata = Metadata::from_str(manifest).unwrap();
         assert!(metadata.proc_macro);
+
+        let manifest = r#"
+            [package]
+            name = "x"
+            [lib]
+            proc_macro = true
+        "#;
+        let metadata = Metadata::from_str(manifest).unwrap();
+        assert!(metadata.proc_macro);
+
+        // Cargo prioritizes `proc-macro` over `proc_macro` in local testing
```

Comment:
That seems like an upstream bug, it should probably give a hard error if they conflict ... but shouldn't affect what docs.rs does.

Reply:
Yep, but there _could_ be existing crates published with it so I wanted to make sure we matched the behavior.

Reply:
And by mentioning it on discord I managed to persuade someone to file an issue 😀 https://github.com/rust-lang/cargo/issues/10299

Reply:
It sounds like this would only warn when you use both at the same time,
 so supporting either key IMO still the right thing to do on docs.rs